src/promptflow/promptflow/executor/_tool_resolver.py

```diff
@@ -264,6 +270,60 @@ def _convert_node_literal_input_types(self, node: Node, tool: Tool, module: type
         updated_node.inputs = updated_inputs
         return updated_node
 
+    def _get_assistant_function_input_types(
```

Comment:
It is not that easy to understand why we have a special logic here, not sharing the logic with tool resolving

Reply:
The orginal resolve method is with Node object. I could merge this method with regular node tool resolving logic, extract something common. The only risk is reorganized orignal tool resolving code. But definitely doable.

Reply:
staled comments